crates/mem-pool/src/fee/types.rs

```diff
@@ -236,15 +240,24 @@ fn parse_l2tx_fee_rate(
             Ok(L2Fee { fee, cycles_limit })
         }
         BackendType::Polyjuice => {
-            // verify the args of a polyjuice L2TX
-            // https://github.com/nervosnetwork/godwoken-polyjuice/blob/aee95c0/README.md#polyjuice-arguments
-            if raw_l2tx_args.len() < (8 + 8 + 16 + 16 + 4) {
-                return Err(anyhow!("Invalid PolyjuiceArgs"));
-            }
+            let poly_args =
+                PolyjuiceParser::from_raw_l2_tx(raw_l2tx).context("parse polyjuice args")?;
             // Note: Polyjuice use CKB_SUDT to pay fee by default
-            let poly_args = raw_l2tx_args.as_ref();
-            let gas_limit = u64::from_le_bytes(poly_args[8..16].try_into()?);
-            let gas_price = u128::from_le_bytes(poly_args[16..32].try_into()?);
+            let (gas_limit, gas_price) = if poly_args.gas_price() > 0 {
+                (poly_args.gas(), poly_args.gas_price())
+            } else {
+                // Gasless tx.
+                rollup_context.check_gasless_tx_to_address(
```

Comment:
IMO we shouldn't reject `gas_price=0` txs here, some networks may accept normal txs which `gas_price=0`.

Reply:
We should return `0` as the gas_price if it is, and only return a gas-less version price when to_address is the entry_point contract.

Reply:
Done.